app/views/api/autocomplete/_show.html.erb

```diff
@@ -4,27 +4,33 @@
       <%= form.label :file, t('dialogs.autocomplete.legend'), class: "govuk-label govuk-label--m" %>
 
       <% if items_present?(@items.component_id) %>
-        <strong class="govuk-warning-text__text govuk-!-margin-bottom-5">
-          <span class="govuk-warning-text__icon govuk-!-margin-left-7" aria-hidden="true">!</span>
-          <span class="govuk-warning-text__assistive">Warning</span>
-          <%= t('dialogs.autocomplete.modal_warning') %>
-        </strong>
+        <div class="govuk-warning-text govuk-!-margin-bottom-2">
+          <span class="govuk-warning-text__icon" aria-hidden="true">!</span>
+          <strong class="govuk-warning-text__text govuk-!-font-weight-regular">
+            <span class="govuk-warning-text__assistive">Warning</span>
+            <%= t('dialogs.autocomplete.modal_warning') %>
+          </strong>
+        </div>
       <% end %>
 
+      <div id="file_hint" class="govuk-hint">
+        <%= t('dialogs.autocomplete.hint') %>
+      </div>
+
       <div class="govuk-form-group <%= @items.errors.present? ? 'govuk-form-group--error' : '' %>">
         <% @items.errors.each do |error|  %>
           <span class="govuk-error-message" id="autocomplete_items_<%=error.attribute%>_error">
             <span class="govuk-visually-hidden">Error:</span><%= error.message %>
           </span>
         <% end %>
 
-        <%= form.file_field :file %>
+        <%= form.file_field :file, class: "govuk-file-upload", 'aria-describedby': 'file_hint autocomplete_items_file_error', accept: ".csv" %>
 
       </div>
-      <p><%= link_to t('dialogs.autocomplete.requirement_link'), '#', class: 'govuk-link govuk-!-margin-bottom-9' %></p>
+      <p class="govuk-!-margin-top-7 govuk-!-margin-bottom-8"><%= link_to t('dialogs.autocomplete.requirement_link'), '#', class: 'govuk-link' %></p>
```

Comment:
I agree that these need to be used with caution, but utility classes like these are useful exactly for situation likes this.  They allow for specific overides where extracting a component is unnecessary.  

Yes CSS is supposed to be write-once/use-many, but it can often end up being write-once/use-once leading to lots of difficult, oddly named, hard to maintain bits of CSS ( and so the CSS file grows... and grows) 

If I hadn't used these then I would have to create a class name and selector to target this one specific paragraph in this one specific dialog, just to add a bit more spacing around it.  

If at some point in the future, we end up resuing this pattern/ markup elsewhere, then it may be worth extracting into its own component css file.